### PR TITLE
Add clamp function for bool tensor

### PR DESCRIPTION
**Pull request by EntongShi2004**
Fixes issue #135048 

Add clamp function for bool tensor in aten\src\ATen\native\cpu\TensorCompareKernel.cpp and aten\src\ATen\native\cuda\TensorCompare.cu, and add some macros in aten\src\ATen\Dispatch.h for two files.

@pytorchbot label "module: core aten"

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @manuelcandales @SherlockNoMad @angelayi

### CODE REVIEW

**Comment by setbit123** [comment]
release notes: module: core aten

**Review by ezyang — CHANGES_REQUESTED** [review]
Get rid of the pytorch submodule. You need to adjust OpInfo tests so this case is tested. I'm not that hot about adding all these extra dispatches. Either what you need already exists, or you should use AT_DISPATCH_V2.

**Comment by ezyang** [comment]
You have some test failures I'd would investigate first

File: test/mobile/model_test/math_ops.py [hunk 1]
```diff
@@ -14,7 +14,6 @@ def pointwise_ops(self):
         b = torch.randn(4)
         t = torch.tensor([-1, -2, 3], dtype=torch.int8)
         r = torch.tensor([0, 1, 10, 0], dtype=torch.int8)
-        t = torch.tensor([-1, -2, 3], dtype=torch.int8)
```
Comment (ezyang) [inline test/mobile/model_test/math_ops.py:17]: ?

Comment (setbit123) [reply test/mobile/model_test/math_ops.py:17]: I found that lines 15 and 17 in the file test/mobile/model_test/math_ops.py are duplicated, so I removed them as a convenience. If you think this needs to be discussed in a new issue or if this approach is something you do not wish to see, I sincerely apologize, and I will restore the content of this line in the PR. Additionally, I have a more urgent matter that I hope you can answer for me, as this has been troubling me for quite some time. I am unsure where to add tests for the clamp function for boolean tensors.

Comment (setbit123) [reply test/mobile/model_test/math_ops.py:17]: cc @ezyang

**Review by ezyang — APPROVED** [review]

**Comment by setbit123** [comment]
Thank you very much for your prompt reply! I can see some failing tests for the PR at https://hud.pytorch.org/pytorch/pytorch/commit/d86ebe2a4691afb970666d99f9b656d7b4b02d99. However, apart from the first one, which I can resolve relatively easily, the remaining errors seem too difficult for me to handle independently. The encouraging part is that the remaining errors, except for the first one, seem to have very similar causes. Could you help me address any of these errors other than the first one? I will follow your approach to resolve the others. Thank you!

cc @ezyang

**Comment by ezyang** [comment]
This is not terribly high priority for me so it will probably be a bit before I take a look
